app/parse.py

```diff
@@ -8,9 +18,54 @@
     tags: list[str]
 
 
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+
+
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag.text for tag in quote_soup.select(".tag")]
+    )
+
+
+def get_next_page(page_soup: BeautifulSoup) -> str:
+    next_page = page_soup.select_one(".pager > li.next > a")
+    if next_page:
+        return next_page["href"]
+
+
+def get_single_page_quotes(page_soup: BeautifulSoup) -> list[Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_quotes() -> list[Quote]:
+    page = requests.get(HOME_URL).content
+    first_page_soup = BeautifulSoup(page, "html.parser")
+
+    next_page = get_next_page(first_page_soup)
+
+    all_quotes = get_single_page_quotes(first_page_soup)
+    while next_page is not None:
+        page = requests.get(urljoin(BASE_URL, next_page)).content
+        soup = BeautifulSoup(page, "html.parser")
+        all_quotes.extend(get_single_page_quotes(soup))
+        next_page = get_next_page(soup)
+    return all_quotes
+
+
+def write_quotes_to_csv(output_csv_path: str, quotes: [Quote]) -> None:
```

Comment:
The type hint for the `quotes` parameter should be `list[Quote]` instead of `[Quote]` to follow Python's type hinting conventions.